astropy/cosmology/funcs.py

```diff
@@ -6,35 +6,93 @@
 import warnings
 import numpy as np
 
+try:
+    from scipy.interpolate import CubicSpline
+except ImportError:
+    HAS_SCIPY = False
+else:
+    HAS_SCIPY = True
```

Comment:
@adrn has centralized the flags for HAS_X. See #11490. When you rebase, these can be imported from ``astropy.utils.compat.optional_deps``